contracts/halo-farm/src/formulas.rs

```diff
@@ -74,46 +85,50 @@ mod test_get_new_reward_ratio_and_time {
 
     #[test]
     fn test_no_staked_token_balance() {
-        let phase_info = get_phase_info();
+        let mut phase_info = get_phase_info();
 
         // No staked token in the farming pool
         let (new_accrued_token_per_share, new_last_reward_time) =
-            phase_info.get_new_reward_ratio_and_time(150, Uint128::zero());
+            phase_info.update_reward_ratio_and_time(150, Uint128::zero());
         assert_eq!(new_accrued_token_per_share, Decimal::zero());
-        assert_eq!(new_last_reward_time, 100);
+        // In this case, last reward time should be updated to current time
+        // for the contract operation, this case only happens when calling deposit()
+        // -> last_reward_time should be updated to current time
+        // Withdraw(), Harvest() or QueryPendingReward will not trigger this case.
+        assert_eq!(new_last_reward_time, 150);
     }
 
     #[test]
     fn test_current_time_before_last_reward_time() {
-        let phase_info = get_phase_info();
+        let mut phase_info = get_phase_info();
 
         // Staked token in the farming pool but current time is before last reward time
         let (new_accrued_token_per_share, new_last_reward_time) =
-            phase_info.get_new_reward_ratio_and_time(50, Uint128::new(100));
+            phase_info.update_reward_ratio_and_time(50, Uint128::new(100));
 
         assert_eq!(new_accrued_token_per_share, Decimal::zero());
```

Comment:
chỗ này phải assert cái phase_info có bị đổi hay ko

Reply:
Added checking phase_info in test
Thank you